cdap-common/bin/functions.sh

```diff
@@ -493,11 +491,34 @@ cdap_set_spark() {
         return 1
       fi
       SPARK_HOME=$(echo -e "${SPARK_VAR_OUT}" | grep ^SPARK_HOME= | cut -d= -f2)
-      export SPARK_HOME
-      return 0
     fi
-    return 1
   fi
+
+  export SPARK_HOME
+
+  # Find environment variables setup via spark-env.sh
+  SPARK_ENV_SH=${SPARK_HOME}/conf/spark-env.sh
```

Comment:
I guess that's fine. I meant getting all of the variables from `SPARK_VAR_OUT` which would mean running `spark-shell` under all cases. However, just reading both possible `spark-env.sh` locations should suffice. People shouldn't be configuring things outside those files, anyway. If you copy what Spark does, itself, I think that's fine, ignoring any other "polluted" environment stuff.